repos/system_upgrade/common/actors/checkfips/actor.py

```diff
@@ -11,25 +12,36 @@ class CheckFips(Actor):
     """
 
     name = 'check_fips'
-    consumes = (KernelCmdline,)
-    produces = (Report,)
+    consumes = (FIPSInfo,)
+    produces = (Report, UpgradeInitramfsTasks)
     tags = (IPUWorkflowTag, ChecksPhaseTag)
 
     def process(self):
-        cmdline = next(self.consume(KernelCmdline), None)
-        if not cmdline:
-            raise StopActorExecutionError('Cannot check FIPS state due to missing command line parameters',
-                                          details={'Problem': 'Did not receive a message with kernel command '
-                                                              'line parameters (KernelCmdline)'})
-        for parameter in cmdline.parameters:
-            if parameter.key == 'fips' and parameter.value == '1':
+        fips_info = next(self.consume(FIPSInfo), None)
+
+        if not fips_info:
+            raise StopActorExecutionError('Cannot check FIPS state due to not receiving necessary FIPSInfo message',
+                                          details={'Problem': 'Did not receive a message with information about FIPS '
+                                                              'usage'})
+
+        if get_target_major_version() == '8':
+            if fips_info.is_enabled:
                 title = 'Cannot upgrade a system with FIPS mode enabled'
-                summary = 'Leapp has detected that FIPS is enabled on this system. ' \
-                          'In-place upgrade of systems in FIPS mode is currently unsupported.'
+                summary = ('Leapp has detected that FIPS is enabled on this system. '
+                           'In-place upgrade of systems in FIPS mode is currently unsupported.')
                 reporting.create_report([
                     reporting.Title(title),
                     reporting.Summary(summary),
                     reporting.Severity(reporting.Severity.HIGH),
                     reporting.Groups([reporting.Groups.SECURITY]),
                     reporting.Groups([reporting.Groups.INHIBITOR])
                 ])
+        else:
+            if fips_info.is_enabled:
+                fips_required_initramfs_files = [
+                    '/etc/crypto-policies/back-ends/opensslcnf.config',
+                    '/etc/pki/tls/openssl.cnf',
+                    '/lib64/.gnutls.hmac',
```

Comment:
For GnuTLS, see also https://bugzilla.redhat.com/show_bug.cgi?id=2148269. It seems they moved the file to a different location that dracut might already expect.

Reply:
I was able to get dracut to automatically pick up the hmac files (`lib64/.libgnutls.so.*.hmac`) by adding the `fips` dracut module when building the upgrade initramfs. Unfortunately, dracut's codebase seems not to mention any of the openssl stuff except for `fips.so`. `fips.so` will be installed with a recently introduced patch  [c5dca3d68915cef077fda2bc5292e12f82cf6dd6](https://github.com/dracutdevs/dracut/commit/c5dca3d68915cef077fda2bc5292e12f82cf6dd6), however, as a part of `crypt` module and there should not be dependency relation between `crypt` and `fips` modules. Therefore, there is nothing "cleaner" leapp can than to specify the openssl files ourselves.

Reply:
Sounds good.